monitoring/loadtest/locust_files/locustfile.py

```diff
@@ -0,0 +1,256 @@
+#!env/bin/python3
+
+import datetime
+import random
+import threading
+import time
+import uuid
+from locust import User, task, between
+from monitoring.monitorlib import auth, infrastructure
+from monitoring.prober.rid import common
+
+
+class DSSClient(infrastructure.DSSTestSession):
+    _locust_environment = None
+
+    def request(self, method, url, **kwargs):
+        if (method == "PUT" and len(url.split("/")) > 3) or method == "PATCH":
+            real_method = "UPDATE"
+        else:
+            real_method = method
+        name = url.split("/")[1]
+        start_time = time.time()
+        result = None
+        try:
+            result = super(DSSClient, self).request(method, url, **kwargs)
+        except Exception as e:
+            self.log_exception(real_method, name, start_time, e)
+        else:
+            if not result or result.status_code != 200:
+                if not result:
+                    msg = "Got None for Response"
+                else:
+                    msg = result.text
+                self.log_exception(real_method, name, start_time, Exception(msg))
+            else:
+                total_time = int((time.time() - start_time) * 1000)
+                self._locust_environment.events.request_success.fire(
+                    request_type=real_method,
+                    name=name,
+                    response_time=total_time,
+                    response_length=0,
+                )
+        return result
+
+    def log_exception(self, real_method, name, start_time, e):
+        total_time = int((time.time() - start_time) * 1000)
+        self._locust_environment.events.request_failure.fire(
+            request_type=real_method,
+            name=name,
+            response_time=total_time,
+            exception=e,
+            response_length=0,
+        )
+
+
+class USS(User):
+    abstract = True
+    isa_dict = {}
+    sub_dict = {}
+
+    def __init__(self, *args, **kwargs):
+        super(USS, self).__init__(*args, **kwargs)
+        oauth_adapter = auth.DummyOAuth("http://localhost:8085/token", "fake_uss")
+        self.client = DSSClient(self.host, oauth_adapter)
+        self.client._locust_environment = self.environment
+        self.client.default_scopes = [common.SCOPE_WRITE, common.SCOPE_READ]
```

Comment:
This results in requesting a token with more permissions than necessary which doesn't follow principle of least privilege best practices; let's leave this None and instead specify `scope=` on each request call ([example](https://github.com/interuss/dss/blob/ca889d11aac4d548a04366ccc58de1753d616ab7/monitoring/prober/rid/test_isa_simple.py#L21)).